boot/mbed/include/mcuboot_config/mcuboot_config.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2018 Open Source Foundries Limited
+ * Copyright (c) 2019-2020 Arm Limited
+ * Copyright (c) 2019-2020 Linaro Limited
+ * Copyright (c) 2020 Embedded Planet
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __MCUBOOT_CONFIG_H__
+#define __MCUBOOT_CONFIG_H__
+
+/*
+ * For available configurations and their explanations,
+ * see mbed_lib.json.
+ */
+
+#define SIGNATURE_TYPE_RSA      0
+#define SIGNATURE_TYPE_EC256    1
+#define SIGNATURE_TYPE_ED25519  2
+
+/*
+ * Signature algorithm
+ */
+#if (MCUBOOT_SIGNATURE_ALGORITHM == SIGNATURE_TYPE_RSA)
+#define MCUBOOT_SIGN_RSA
+#  if (MCUBOOT_RSA_SIGNATURE_LENGTH != 2048 && \
+       MCUBOOT_RSA_SIGNATURE_LENGTH != 3072)
+#    error "Invalid RSA key size (must be 2048 or 3072)"
+#  else
+#    define MCUBOOT_SIGN_RSA_LEN MCUBOOT_RSA_SIGNATURE_LENGTH
+#  endif
+#elif (MCUBOOT_SIGNATURE_ALGORITHM == SIGNATURE_TYPE_EC256)
+#define MCUBOOT_SIGN_EC256
+#elif (MCUBOOT_SIGNATURE_ALGORITHM == SIGNATURE_TYPE_ED25519)
+#define MCUBOOT_SIGN_ED25519
+#endif
+
+/*
+ * Crypto backend
+ */
+#if (MCUBOOT_CRYPTO_BACKEND == MBEDTLS)
+#define MCUBOOT_USE_MBED_TLS
+#elif (MCUBOOT_CRYPTO_BACKEND == TINYCRYPT)
+#error TinyCrypt backend is not supported until RNG from Mbed OS is integrated
```

Comment:
It is used for signature verification and ECIES using secp256r1 and AES encryption; RNG should not be required.

Reply:
[This header](https://github.com/mcu-tools/mcuboot/blob/master/ext/tinycrypt/lib/include/tinycrypt/ecc_platform_specific.h#L77) sets `#define default_RNG_defined 1` which tells source files trying to use `default_CSPRNG()` which is undefined. This is the reason we initially marked TinyCrypt as unsupported. Could we change it to 0?

Reply:
> Could we change it to 0?

No, but I would suggest you try something like this on mcuboot_main.cpp:

```
#if (MCUBOOT_CRYPTO_BACKEND == TINYCRYPT)
/* XXX add this global definition for linking only */
int default_CSPRNG(uint8_t *dest, unsigned int size) { return 0; }
#endif

int main() 
{
    ...

    /* XXX add this close to Mbed TLS initialization */
    #if (MCUBOOT_CRYPTO_BACKEND == TINYCRYPT)
        uECC_set_rng(0);
    #endif

    ...
```

Reply:
That's a good idea

Reply:
I will implement the changes above however I'm wondering if there are any impacts on security if TinyCrypt expects a TRNG and there is not one available. Is this workaround purely to satisfy missing symbols? From what @utzig said, mcuboot appears to use TinyCrypt in a way that it will not call TRNG functions.

Reply:
> I will implement the changes above however I'm wondering if there are any impacts on security if TinyCrypt expects a TRNG and there is not one available

MCUboot only calculates checksum, where no randomness is involved, so there should be no security implication as far as I understand. RNG is more useful for key generation which is not done by our bootloader.

Reply:
I did some re-check of the Tinycrypt code. For signature validation the TRNG really is not needed as one would expect, so no issues; for encryption it is used to blind the montgomery ladder, so it has security implications, because not using it would allow someone to use differential power analysis to recover a secret when doing the ECDH. But honestly this is just from a fast inspection, I have to think this more thoroughly; unfortunately today I don't have the time. If you are not using encrypted images you are totally fine!

Reply:
Integration with Mbed OS TRNG should be fairly simple, the API is https://github.com/ARMmbed/mbed-os/blob/master/hal/include/hal/trng_api.h
But it can be a future addition in a separate PR, if we want to enable encryption at some point.

Reply:
I will make sure to add a note regarding this in the PR commit message.